test/unit/org/apache/cassandra/db/compaction/CompactionStrategyManagerPendingRepairTest.java

```diff
@@ -371,15 +365,6 @@ public void testFinalizedAndCompactionRace() throws IOException
         assertEquals(1, cfs.getLiveSSTables().size());
         assertEquals(1, cfs.getPendingRepairSSTables(repairID).size());
 
-        System.out.println("*********************************************************************************************");
-        System.out.println(compactedSSTable);
-        System.out.println("Pending repair UUID: " + compactedSSTable.getPendingRepair());
-        System.out.println("Repaired at: " + compactedSSTable.getRepairedAt());
-        System.out.println("Creation time: " + compactedSSTable.getCreationTimeFor(Component.DATA));
-        System.out.println("Live sstables: " + cfs.getLiveSSTables().size());
-        System.out.println("Pending repair sstables: " + cfs.getPendingRepairSSTables(repairID).size());
-        System.out.println("*********************************************************************************************");
-
         // Run compaction again. It should pick up the pending repair sstable and issue a RepairFinishedCompactionTask
         compactionTasks = compactionStrategyContainer.getNextBackgroundTasks(FBUtilities.nowInSeconds());
```

Comment:
This one, on the other hand, could return nothing if the problem is solved differently. We should wrap the next checks and executing the task in an `if (!compactionTasks.isEmpty())`.

Reply:
Right. If the compaction strategy actively detected the situation and fixed it without the need of returning the tasks. Added.